subsys/debug/Kconfig

```diff
@@ -163,6 +163,19 @@ config ASSERT_LEVEL
 	  Level 1: on + warning in every file that includes __assert.h
 	  Level 2: on + no warning
 
+config SPIN_VALIDATE
+	bool "Enable spinlock validation"
+	depends on ASSERT
+	default y if MP_NUM_CPUS < 4 && (!FLASH || FLASH_SIZE > 32)
```

Comment:
The MP_NUM_CPUS is a hard requirement, not a configuration default.  It needs to stay in the C code.  (The reason is that the implementation using two bits in the bottom of a pointer union to store a CPU index, which only has space for four CPUs).

And regarding the size check:  IIRC it was only ~two platforms that ran into code size trouble with this.  It would probably be easier to just default this to "y" and change just the tiny failing platforms to disable it specifically (with a comment like "too big for our flash size").  That's probably how it should have been done originally, but at the time it wasn't available via kconfig; now it is. :)

Reply:
Ok, I see, thank you for the feedback. I will move it back to the C code with BUILD_ASSERT, as you suggest below. That way we will not end up with at Kconfig option which, when enabled, ends up being silently disabled.

As I do not know which specific platforms were affected by the size problem, I would suggest to leave the default check in the Kconfig, as not to cause unnecessary problems for those platforms.

Reply:
I decided to leave the MP_NUM_CPUS check as a 'depends on' as that hopefully leads to less build assert hits and as it makes it clearer from a Kconfig point of view, that it depends upon the number of cpus. Hope you agree.